src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,190 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import cn from 'classnames';
+
+import { Todo } from '../../types/Todo';
+import { ErrorMessage, TodosContext } from '../TodosContext';
+import { updateTodo } from '../../api/todos';
+
+type Props = {
+  todo: Todo,
+};
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const {
+    id,
+    title,
+    completed,
+  } = todo;
+
+  const {
+    handleRemoveTodo,
+    setAlarm,
+    setIsTodoChange,
+    changingItems,
+    setChangingItems,
+    setTodos,
+  } = useContext(TodosContext);
+
+  const [isCompleted, setIsCompleted] = useState(completed);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isUpdating, setIsUpdating] = useState(false);
+  const [editedInput, setEditedInput] = useState(title);
+
+  const editedFormInput = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (isUpdating && editedFormInput.current) {
+      editedFormInput.current.focus();
+    }
+  }, [isUpdating]);
+
+  useEffect(() => {
+    setIsCompleted(completed);
+  }, [completed]);
+
+  const handleUpdateTodo = (val: boolean | string) => {
+    let newTodo = todo;
+
+    if (typeof val === 'boolean') {
+      newTodo = {
+        ...todo,
+        completed: val,
+      };
+    }
+
+    if (typeof val === 'string' && val.trim() !== '') {
+      newTodo = {
+        ...todo,
+        title: val,
+      };
+    }
+
+    if (newTodo) {
```

Comment:
```suggestion
  const handleUpdateTodo = (updateFields: Partial<Todo>) => {
    const updatedTodo = {
      ...todo,
      ...updateFields,
    };
```
You can pass directly the fields to update and in this case, you don't need such strange checks. Something like this.
```javacript
handleUpdateTodo({ completed: !isCompleted });
```

Also, you always have todo so the last check is redundant.